mailet/base/src/test/java/org/apache/mailet/base/test/FakeMailContext.java

```diff
@@ -44,11 +49,17 @@
         private final MailAddress sender;
         private final Collection<MailAddress> recipients;
         private final MimeMessage msg;
+        private final List<String> attributeNames;
 
-        public SentMail(MailAddress sender, Collection<MailAddress> recipients, MimeMessage msg) {
+        public SentMail(MailAddress sender, Collection<MailAddress> recipients, MimeMessage msg, List<String> attributeNames) {
             this.sender = sender;
-            this.recipients = recipients;
+            this.recipients = ImmutableList.copyOf(recipients);
```

Comment:
I reproduced it again and other again :

```
Expecting:
 <[SentMail{recipients=[sender@domain.com], sender=null, attributeNames={}}]>
to contain only:
 <[SentMail{recipients=[sender@domain.com], sender=null, attributeNames={}}]>
elements not found:
 <[SentMail{recipients=[sender@domain.com], sender=null, attributeNames={}}]>
and elements not expected:
 <[SentMail{recipients=[sender@domain.com], sender=null, attributeNames={}}]>
```

After little investigations it appeared the mailAddress collection was not equal...